pkg/initiailizer_v2/utils/utils.py

```diff
@@ -0,0 +1,37 @@
+import os
+from abc import ABC, abstractmethod
+from dataclasses import fields
+from typing import Dict
+
+STORAGE_URI_ENV = "STORAGE_URI"
+HF_SCHEME = "hf"
+
+
+class ModelProvider(ABC):
+    @abstractmethod
+    def load_config(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def download_model(self):
+        raise NotImplementedError()
+
+
+class DatasetProvider(ABC):
+    @abstractmethod
+    def load_config(self):
+        raise NotImplementedError()
+
+    @abstractmethod
+    def download_dataset(self):
+        raise NotImplementedError()
+
+
+# Get DataClass config from the environment variables.
+# Env names must be equal to the DataClass parameters.
+def get_config_from_env(config) -> Dict[str, str]:
```

Comment:
do you want some typing hints for config

Reply:
Ah I see. Then it’s fine to go without.

Reply:
I’m not much of a Python dev these days so I don’t know about the typing for data classes.

Reply:
shall we use like this
`config : Union[DataClassA, DataClassB]`

Reply:
I tried this, it won't work. E.g. I can see this error from PyLance:
```
Argument of type "type[HuggingFaceDatasetConfig]" cannot be assigned to parameter "config" of type "HuggingFaceDatasetConfig" in function "get_config_from_env"
  "type[type]" is not assignable to "type[HuggingFaceDatasetConfig]"
  ```

Reply:
@deepanker13 What type validator do you use in your IDE ?
My VSCode complains with the error that I added above.
I am using PyLance VSCode extension: https://github.com/microsoft/pylance-release